device/daita.go

```diff
@@ -0,0 +1,301 @@
+//go:build daita
+// +build daita
+
+package device
+
+import (
+	"time"
+	"unsafe"
+)
+
+// #include <stdio.h>
+// #include <stdlib.h>
+// #include "../cmaybenot/cmaybenot.h"
+// #cgo LDFLAGS: -L${SRCDIR}/../cmaybenot/target/release -l:libcmaybenot.a -lm
+import "C"
+
+type MaybenotDaita struct {
+	events         chan *Event
+	actions        chan *Action
+	maybenot       *C.Maybenot
+	newActionsBuf  []C.MaybenotAction
+	machineActions map[uint64]Action
+	logger         *Logger
+}
+
+type Event struct {
+	// The machine that generated the action that generated this event, if any.
+	Machine uint64
+
+	Peer      NoisePublicKey
+	EventType EventType
+	XmitBytes uint16
+}
+
+type ActionType uint32
+
+const (
+	Cancel        = ActionType(0)
+	InjectPadding = ActionType(1)
+	BlockOutgoing = ActionType(2)
+)
+
+const (
+	ERROR_GENERAL_FAILURE      = -1
+	ERROR_INTERMITTENT_FAILURE = -2
+)
+
+type EventContext struct {
+	tunnelHandle int32
```

Comment:
What's this?

Reply:
I think this is legacy? the victim of refactoring.